app/services/appointment_scheduler.rb

```diff
@@ -37,14 +37,12 @@ def schedule(patient:, ubs_id:, from:, reschedule:)
       )
       return [NO_SLOTS] unless success
 
-      if reschedule
-        new_appointment = patient.appointments.waiting.last!
-      else
-        new_appointment = patient.appointments.waiting.where.not(id: current_appointment&.id).first!
-      end
+      new_appointment = patient.appointments.order(:updated_at).last!
 
       cancel_schedule(appointment: current_appointment, new_appointment: new_appointment) if current_appointment
 
+      patient.doses.order(:sequence_number).last.update!(follow_up_appointment: new_appointment) if patient.doses.exists?
```

Comment:
You could do something like this on `appointments_controller`

Reply:
Maybe, but make sense for my update the information dose at scheduler scope